packages/stex-react-renderer/src/lib/ProblemDisplay.tsx

```diff
@@ -406,27 +406,46 @@ function groupingByBloomDimension(data?: string) {
   return groupedData;
 }
 
-function DimAndURIListDisplay({ title, data }: { title: string; data?: string }) {
-  const groupedData = groupingByBloomDimension(data);
+function isBloomDimensionFormat(data: string) {
+  const [dimension, _] = data.split(':');
+  return Object.values(BloomDimension)
+    .map((value) => value.toLowerCase())
+    .includes(dimension.toLowerCase());
+}
+
+function structureData(data?: string): Record<string, string[]> {
+  if (!data) return {};
+  if (isBloomDimensionFormat(data)) {
+    return groupingByBloomDimension(data);
+  }
+  return {
+    NON_BLOOM_URIS: data.split(',').map((uri) => decodeURIComponent(uri.trim())),
+  };
+}
+
+export function DimAndURIListDisplay({ title, data }: { title: string; data?: string }) {
```

Comment:
Your logic has become very convoluted. I was suggesting this approach yesterday.
1. Create a component that displays uris
export function URIListDisplay({ uris }: { uris?: string[] }) {
```
return uris.map((uri, index, array) => (
              <React.Fragment key={index}>
                <span>{mmtHTMLToReact(getMMTHtml(uri))}</span>
}
```
2. Use the above component is DimAndUriListDisplay:
export function DimAndURIListDisplay({ title, data }: { title: string; data?: string }) {
```
export function DimAndURIListDisplay({ title, data }: { title: string; data?: string }) {
const transformedData = structureData(data);
  return (
    <Box border="1px solid black" mb="10px" bgcolor="white">
      <Typography fontWeight="bold" sx={{ p: '10px' }}>
        {title}&nbsp;
      </Typography>
       {Object.entries(transformedData)
        .filter(([_, uris]) => uris.length > 0)
        .map(([group, uris]) => (
          <Box key={group} borderTop="1px solid #AAA" p="5px" display="flex" flexWrap="wrap">
            {group !== 'NON_BLOOM_URIS' && <DimIcon dim={group as BloomDimension} />} &nbsp;
            <URIListDisplay uris={uris} />
  ))}
          </Box>
              </Box>
  );
}

3. Use the component `URIListDisplay` directly for things like `crossrefs`